### ISSUE TITLE
BUG/Q: Multiple "not equal" comparisons on one argument in one call to seccomp_rule_add() do not generate a proper filter

### ISSUE THREAD

**Issue by varqox** [seccomp/libseccomp#118]
I want to add filter that allows fstat only if fd is not 0, 1 or 2. I cannot get it to work. As long as I want to restrict it only on one fd (e.g. 0) all works fine, but with more it does not. Here is some demonstrative code: 
```c
#include <seccomp.h>
#include <stdio.h>
#include <errno.h>

#define _GNU_SOURCE
#include <sys/syscall.h>
#include <unistd.h>

int main() {
	scmp_filter_ctx ctx = seccomp_init(SCMP_ACT_KILL);
	if (ctx == NULL) {
		perror("seccomp_init()");
		return 1;
	}

	seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(write), 0);

	// I want to disallow fstat() on stdin, stdout and stderr (for demonstrative purposes)
	seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EPERM), SCMP_SYS(fstat), 1, SCMP_A0(SCMP_CMP_EQ, 0));
	seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EPERM), SCMP_SYS(fstat), 1, SCMP_A0(SCMP_CMP_EQ, 1));
	seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EPERM), SCMP_SYS(fstat), 1, SCMP_A0(SCMP_CMP_EQ, 2));

	seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(fstat), 3,
		SCMP_A0(SCMP_CMP_NE, 0),
		SCMP_A0(SCMP_CMP_NE, 1),
		SCMP_A0(SCMP_CMP_NE, 2));

	int stdout_fileno = 1;
	seccomp_export_pfc(ctx, stdout_fileno);

	seccomp_load(ctx);

	// Confirms that the rule was not generated properly
	write(1, ">>> 1\n", 6);
	syscall(SYS_fstat, 1, NULL);
	write(1, ">>> 2\n", 6);
	syscall(SYS_fstat, 4, NULL);
	write(1, ">>> 3\n", 6); // Should reach here

	return 0;
}
```

**Comment by varqox** [comment]
Thank you for help.
I cannot recall reading about that in the documentation. Maybe it is a good idea to mention that constraint there. Also adding some information how multiple rules on one syscall work together within libseccomp. Deducing it from the output of seccomp_export_pfc() sometimes gives unexpected results. Such specification would be helpful to others.

**Comment by pcmoore** [comment]
Closed with PR #225.
